lib/reesmat.gi

```diff
@@ -123,6 +123,11 @@ InstallMethod(IsSimpleSemigroup,
 [IsReesMatrixSubsemigroup and HasUnderlyingSemigroup],
 R-> IsSimpleSemigroup(UnderlyingSemigroup(R)));
 
+InstallMethod(IsSimpleSemigroup, 
```

Comment:
The problem with this is that if somebody sees this code in the future, they may wonder "huh, why is this duplicated?" and remove the new method again. I see two courses to protected against that:
1. Add a comment explaining the reason,
2. don't add this new method, instead adjust the rank of the existing method, by using a dynamic rank adjustment via  `{} -> RankFilter(IsFinite)`.

Reply:
This PR initially implemented item 2: I added the line `{} -> RankFilter(IsFinite)`. After @stevelinton's comment https://github.com/gap-system/gap/pull/2989#issuecomment-437803909 I changed it to how it is now. Therefore I'll implement your suggestion 1.